### DISCUSSION TITLE
Adding VR to CARLA

### DISCUSSION THREAD

**Discussion by Piti98** [carla-simulator/carla#5667] [Show and tell]
Hello everyone,

as part of a experiment for a university project, I found a way to add VR support to Carla and to create a first person driving experience.
It is not perfect yet, but rather a baseline to start from, so any suggestions for improving and expanding are welcome.

I tried to describe the whole process as detailed as possible, so feel free to try it out for youself: [Adding VR to CARLA .pdf](https://github.com/carla-simulator/carla/files/9333229/Adding.VR.to.CARLA.pdf)

Have fun!

**Comment by LeonardoVezzani** [comment]
This is really helpful; can you provide the specs of the setup used for testing?

**Reply by Piti98** [reply]
Glad it's useful! For this project I used my gaming computer from home. The hardware is no longer high-tech, but still solid.

- GPU: Nvidia GeForce GTX 1080 Ti
- CPU: Intel i7-4790K @ 4.4 GHz
- RAM: 16 GB
- VR headset: Oculus Rift S

As already written in the documentation, the performance was rather poor and very stuttering. But it might improve with better hardware and simplifications of the environment (lights, textures etc.).

**Reply by LeonardoVezzani** [reply]
Unfortunatly, even with a 3060 or 3070 + quest2, its quite unusable as is. but I believe there is ample room for improvement

**Reply by Zhenlin-Xu** [reply]
Thank you for sharing your insights and guidance. Do you find other approaches to run CARLA in VR modes?

**Reply by LeonardoVezzani** [reply]
Unfortunately not, but I'm currently working on developing my own with all immersive features such as dashboards.
If you are interested contact me we could find some area to collaborate

**Comment by leepms** [comment]
thank you for your great work and share spirit, 
your technical guidance give me valuable advices and guidance.
if you have more experirent about this work, may I have a chance chat about this with you?

**Comment by xavisolesoft** [comment]
Thank you for sharing!

**Comment by SimonSaysGiveMeSmile** [comment]
Remember to include your name in the Technical Guide, G, thank you for the good work!

Update: I followed the guide yet when I try to "Play" the simulation, the "VR Preview" is grayed out. Could it be a settings issue?
